collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
+
+    def __get__(self, instance, owner):
```

Comment:
You mean **in place of** descriptors right? I think that the right way is to have a getter and setter with the right behavior implemented. Descriptors gets us that without verbosity of declaring properties. Its focused on just the impacted properties. 

We can use getattr and setattr, but then we pay the penalty of the check for conversion on each attribute access. Composition vs Inheritance as a design principle. 

Why does getattr, setattr of the entire object feel more correct than a get, set on the attribute itself?